### ISSUE TITLE
Distinguish between Double and Long/Integer in generated code

### ISSUE THREAD

**Issue by jeromesimeon** [accordproject/jura#37]
There are some issues in how numeric values are being handled. The compiler doesn't produce code which properly distinguish operations on Double and operations on Integer(or Long).

**Comment by jeromesimeon** [comment]
Here is an example of wrong code-generation:
```
bash-3.2$ cat examples/volumediscount/logic-err.jura
cat examples/volumediscount/logic-err.jura
namespace org.accordproject.volumediscount

import org.accordproject.contract.*
import org.accordproject.late.*

contract VolumeDiscount over TemplateModel {
    clause volumediscount(request VolumeDiscountRequest) : VolumeDiscountResponse {
      new VolumeDiscountResponse{ discountRate: 0.1 }
  }
}
bash-3.2$ jurac execute --jura ./examples/volumediscount/logic-err.jura --contractname VolumeDiscount --clausename volumediscount --clause ./examples/volumediscount/clause.json --request ./examples/volumediscount/request.json --cto ./examples/volumediscount/model.cto 
jurac execute --jura ./examples/volumediscount/logic-err.jura --contractname VolumeDiscount --clausename volumediscount --clause ./examples/volumediscount/clause.json --request ./examples/volumediscount/request.json --cto ./examples/volumediscount/model.cto 
14:22:00 - info: Logging initialized. 2018-03-19T18:22:00.580Z
14:22:00 - info: {"discountRate":"0.1","$class":"org.accordproject.volumediscount.VolumeDiscountResponse"}
bash-3.2$ jurac compile --jura ./examples/volumediscount/logic-err.jura --cto ./examples/volumediscount/model.cto 
jurac compile --jura ./examples/volumediscount/logic-err.jura --cto ./examples/volumediscount/model.cto 
14:22:13 - info: Logging initialized. 2018-03-19T18:22:13.021Z
14:22:13 - info: 'use strict';
/*eslint-disable no-unused-vars*/
/*eslint-disable no-undef*/
/*eslint-disable no-var*/


class VolumeDiscount {
  volumediscount(context) {
    return brand(["org.accordproject.volumediscount.VolumeDiscountResponse"],{"discountRate": "0.1"});
  }
}

/*eslint-enable no-unused-vars*/
/*eslint-enable no-undef*/
```

**Comment by jeromesimeon** [comment]
Some initial fix based on a not-yet released branch of Q*cert for native handling of floating point numbers can be found in https://github.com/accordproject/jura/tree/flocq

With that branch, compilation and execution of the above example gives the right result:
```
bash-3.2$ jurac execute --jura ./examples/volumediscount/logic-err.jura --contractname VolumeDiscount --clausename volumediscount --clause ./examples/volumediscount/clause.json --request ./examples/volumediscount/request.json --cto ./examples/volumediscount/model.cto 
jurac execute --jura ./examples/volumediscount/logic-err.jura --contractname VolumeDiscount --clausename volumediscount --clause ./examples/volumediscount/clause.json --request ./examples/volumediscount/request.json --cto ./examples/volumediscount/model.cto 
15:26:27 - info: Logging initialized. 2018-03-19T19:26:27.068Z
15:26:27 - info: {"discountRate":0.1,"$class":"org.accordproject.volumediscount.VolumeDiscountResponse"}
```

**Comment by jeromesimeon** [comment]
Fixed by 1aee577a03a4590182ae844d29695e9f7041b603 and d9ebfa526b76eb032b799c1550534c7d62d8e6f4.
